socs/agents/scpi_psu/agent.py

```diff
@@ -48,9 +53,22 @@ def init(self, session, params=None):
                 self.log.error(f"PSU timed out during connect: {e}")
                 return False, "Timeout"
             self.log.info("Connected to psu: {}".format(self.idn))
-
         return True, 'Initialized PSU.'
 
+    def _initialize_module(self, session, params=None):
+        """Initialize the ScpiPsu module."""
+        try:
+            self.psu = PsuInterface(self.ip_address, self.gpib_slot)  # this is only necessary if the ip_address or gpib_slot has changed. I could declare a self.connected flag if this is uneeded.
+            self.idn = self.psu.identify()
+        except Exception as e:  # socket.tiout OR OSError 113 No route to host.
+            self.log.info("Failed to reconnect, trying again")
```

Comment:
Two things here -- I'd change `self.log.info` to `self.log.warn`, and also log the error encountered, either in a separate line or like:
```
self.log.warn(f"Error establishing connection: {e}")
```